pandas/tests/resample/test_datetime_index.py

```diff
@@ -149,29 +147,28 @@ def _ohlc(group):
 
         inds = date_range('1/1/2000', periods=4, freq='5min', name='index')
 
-        for arg in args:
+        if arg == 'ohlc':
+            func = _ohlc
+        else:
+            func = arg
+        try:
+            result = getattr(s.resample(
+                '5min', closed='right', label='right'), arg)()
+
+            expected = s.groupby(grouplist).agg(func)
+            assert result.index.name == 'index'
             if arg == 'ohlc':
-                func = _ohlc
+                expected = DataFrame(expected.values.tolist())
+                expected.columns = ['open', 'high', 'low', 'close']
+                expected.index = Index(inds, name='index')
+                assert_frame_equal(result, expected)
             else:
-                func = arg
-            try:
-                result = getattr(s.resample(
-                    '5min', closed='right', label='right'), arg)()
-
-                expected = s.groupby(grouplist).agg(func)
-                assert result.index.name == 'index'
-                if arg == 'ohlc':
-                    expected = DataFrame(expected.values.tolist())
-                    expected.columns = ['open', 'high', 'low', 'close']
-                    expected.index = Index(inds, name='index')
-                    assert_frame_equal(result, expected)
-                else:
-                    expected.index = inds
-                    assert_series_equal(result, expected)
-            except BaseException as exc:
-
-                exc.args += ('how=%s' % arg,)
-                raise
+                expected.index = inds
+                assert_series_equal(result, expected)
+        except BaseException as exc:
```

Comment:
can you remove this catching of BaseException (e.g. the try/accept), with parameterized methods it is uncessary